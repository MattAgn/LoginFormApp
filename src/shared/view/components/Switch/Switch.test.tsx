import { screen, userEvent } from '@testing-library/react-native';
import { useState } from 'react';

import { Switch } from '#shared/view/components/Switch/Switch';

import { renderWithProviders } from '#testing/render';

const TestSwitch = () => {
  const [isChecked, setIsChecked] = useState(false);

  return (
    <Switch
      isChecked={isChecked}
      onPress={() => setIsChecked(!isChecked)}
      accessibilityLabel="switch"
    />
  );
};

describe('Switch', () => {
  it('renders properly', async () => {
    renderWithProviders(<TestSwitch />);

    expect(screen).toMatchComponentSnapshot();
    expect(screen.getByRole('switch')).toHaveAccessibilityState({
      checked: false,
    });

    await userEvent.press(screen.getByRole('switch'));

    expect(screen).toMatchComponentSnapshot();
    expect(screen.getByRole('switch')).toHaveAccessibilityState({
      checked: true,
    });
  });
});
