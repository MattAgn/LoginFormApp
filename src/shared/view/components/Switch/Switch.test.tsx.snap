// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch renders properly 1`] = `null`;

exports[`Switch renders properly 2`] = `
<RNCSafeAreaProvider
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
  >
    <View
      accessibilityLabel="switch"
      accessibilityRole="switch"
      accessibilityState={
        {
          "checked": false,
        }
      }
      accessible={true}
      style={
        {
          "backgroundColor": undefined,
          "borderColor": undefined,
          "borderRadius": 20.5,
          "borderWidth": 1,
          "height": 41,
          "justifyContent": "center",
          "width": 74,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row-reverse",
            "height": "100%",
            "justifyContent": "space-between",
          }
        }
      >
        <View
          style={
            {
              "width": 10,
            }
          }
        />
        <Text
          allowFontScaling={false}
          layout={BaseAnimationMock {}}
          numberOfLines={0}
          style={
            {
              "color": undefined,
              "fontFamily": "OpenSans-Regular",
              "fontSize": 12,
              "lineHeight": 18,
              "textDecorationLine": "none",
              "textTransform": "uppercase",
            }
          }
        />
        <View
          style={
            {
              "flex": 1,
            }
          }
        />
        <View
          layout={BaseAnimationMock {}}
          style={
            {
              "alignItems": "center",
              "backgroundColor": undefined,
              "borderColor": undefined,
              "borderRadius": 15.5,
              "borderWidth": 1,
              "height": 31,
              "justifyContent": "center",
              "width": 31,
            }
          }
        />
        <View
          style={
            {
              "width": 5,
            }
          }
        />
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;

exports[`Switch renders properly 3`] = `null`;

exports[`Switch renders properly 4`] = `
<RNCSafeAreaProvider
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
  >
    <View
      accessibilityLabel="switch"
      accessibilityRole="switch"
      accessibilityState={
        {
          "checked": true,
        }
      }
      accessible={true}
      style={
        {
          "backgroundColor": undefined,
          "borderColor": undefined,
          "borderRadius": 20.5,
          "borderWidth": 1,
          "height": 41,
          "justifyContent": "center",
          "width": 74,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "height": "100%",
            "justifyContent": "space-between",
          }
        }
      >
        <View
          style={
            {
              "width": 10,
            }
          }
        />
        <Text
          allowFontScaling={false}
          layout={BaseAnimationMock {}}
          numberOfLines={0}
          style={
            {
              "color": undefined,
              "fontFamily": "OpenSans-Regular",
              "fontSize": 12,
              "lineHeight": 18,
              "textDecorationLine": "none",
              "textTransform": "uppercase",
            }
          }
        />
        <View
          style={
            {
              "flex": 1,
            }
          }
        />
        <View
          layout={BaseAnimationMock {}}
          style={
            {
              "alignItems": "center",
              "backgroundColor": undefined,
              "borderColor": undefined,
              "borderRadius": 15.5,
              "borderWidth": 1,
              "height": 31,
              "justifyContent": "center",
              "width": 31,
            }
          }
        />
        <View
          style={
            {
              "width": 5,
            }
          }
        />
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
